src/safeposix/syscalls/net_calls.rs

```diff
@@ -1295,14 +1291,14 @@ impl Cage {
     }
 
     //we only return the default host name because we do not allow for the user to change the host name right now
-    pub fn gethostname(self, length: usize, mut address_ptr: &[u8]) -> i32 {
+    pub fn gethostname(self, length: usize, address_ptr: &mut [u8]) -> i32 {
         if length < 0 {
             return syscall_error(Errno::EINVAL, "gethostname", "invalid argument");
         }
         if DEFAULT_HOSTNAME.chars().count() > length {
-            address_ptr = &DEFAULT_HOSTNAME[..length].as_bytes();
+            address_ptr.copy_from_slice(&DEFAULT_HOSTNAME[..length].as_bytes());
         } else {
-            address_ptr = &*DEFAULT_HOSTNAME.as_bytes();
+            address_ptr.copy_from_slice(&DEFAULT_HOSTNAME.as_bytes());
```

Comment:
As far as I know, for copy_from_slice, both slices have to be the same size, and this is enforced at runtime. You'll need to get a subslice of address pointer to ensure that.